### PR TITLE
cmake: link mbedtls if available

### PR DESCRIPTION
**Pull request by cspiel1**

### CODE REVIEW

**Comment by sreimers** [comment]
> * in both, and how should we avoid a duplicate?

I think it has to be placed in both. Since we should not install a generic `FindMBEDTLS.cmake` with re.

**Comment by cspiel1** [comment]
MbedTLS can also be built as shared library with `USE_SHARED_MBEDTLS_LIBRARY=on`.  I try to set `STATIC` of baresip automatically.

**Comment by alfredh** [comment]
I would like to suggest that we get this enabled in libre first, and get it working there.

After that we can update the applications.

**Comment by alfredh** [comment]
is this needed?

re can link to mbedtls and baresip to openssl

**Comment by cspiel1** [comment]
Yes I think so. We are going to evaluate baresip/re on Zephyr. So there is no openssl.

**Comment by alfredh** [comment]
building with STATIC=1 and static libre.a is not related.

1. A library can be static archive (.a) or dynamic (.so)
2. Baresip does not need to know if we are using static/dynamic mbedtls
3. If libre.a is linked statically, it does not provide any info about which libs to link to


In theory it should be possible to build libre with OpenSSL and baresip with MbedTLS, or vice versa.

File: CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,10 @@ else()
   option(STATIC "Build static" OFF)
 endif()
 
+find_package(MBEDTLSSO)
+if(USE_MBEDTLS AND NOT MBEDTLSSO_FOUND)
+  set(STATIC BOOL ON)
```
Comment (alfredh) [inline CMakeLists.txt:54]: I dont agree with this logic

File: cmake/FindMBEDTLSSO.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
+find_library(MBEDTLSSO_LIBRARY
```
Comment (alfredh) [inline cmake/FindMBEDTLSSO.cmake:1]: I dont think there is a need for 2 different FindMbedTLS files -- one should be enough

**Comment by cspiel1** [comment]
You're right. Tried without linking to mbedtls in zephyr and it worked. Since everything is linked statically, we don't need this PR.

Shared linking was only meant for developing on the dev host. But this is not needed.

**Pull request closed without merging**